### PR TITLE
Job assign at roundstart now favors empty positions first

### PR DESCRIPTION
**Pull request by ChrisDirksDev**
:cl: Crexfu
tweak: Job assignment at roundstart now favors empty positions first
:cl:

[why]: # (Please add a short description [on the next line] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding:) 

Currently in low pop rounds multiple players can be put into the same job leaving other jobs empty that they would otherwise be happy to roll. This change seeks to improve the user experience in these situations  by better distributing players. It may be worth having a preference for wanting to be filled into empty roles  first so that the social butterflies can opt out and more likely be grouped with others in low pop rounds.

### CODE REVIEW

**Comment by Jalleo** [comment]
This may need testing to make sure it works properly and does not unbalance things like Revolution rounds.

File: code/controllers/subsystem/job.dm [hunk 1]
```diff
@@ -308,16 +310,24 @@ SUBSYSTEM_DEF(job)
...
-						Debug("DO pass, Player: [player], Level:[level], Job:[job.title]")
-						AssignRole(player, job.title)
-						unassigned -= player
-						break
+
+						validjob = job
+
+						//Is the Job empty? Stop Looking Then!
+						if (job.current_positions == 0)
+							break;
```
Comment (Cyberboss) [inline code/controllers/subsystem/job.dm]: no need for semicolons outside defines

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/controllers/subsystem/job.dm [hunk 2]
```diff
@@ -308,16 +310,24 @@ SUBSYSTEM_DEF(job)
...
 					if((job.current_positions < job.spawn_positions) || job.spawn_positions == -1)
-						Debug("DO pass, Player: [player], Level:[level], Job:[job.title]")
-						AssignRole(player, job.title)
-						unassigned -= player
-						break
+
+						validjob = job
+
+						//Is the Job empty? Stop Looking Then!
+						if (job.current_positions == 0)
```
Comment (lzimann) [inline code/controllers/subsystem/job.dm]: just `!job.current_positions` will do

File: code/controllers/subsystem/job.dm (hunk 1 above)
Comment (lzimann) [inline code/controllers/subsystem/job.dm]: no commas, please

**Comment by Cyberboss** [comment]
Does this account for selection priority? because that should come first

File: code/controllers/subsystem/job.dm (hunk 1 above)
Comment (Cyberboss) [inline code/controllers/subsystem/job.dm]: >commas

Comment (lzimann) [inline code/controllers/subsystem/job.dm]: \>english words

**Comment by lzimann** [comment]
IIRC this is in the select a random job part. It should not throw off any job priority.

**Comment by Crexfu** [comment]
yes it is within the priority loop from high to low

File: code/controllers/subsystem/job.dm (hunk 1 above)
Comment (Crexfu) [inline code/controllers/subsystem/job.dm]: corrected

File: code/controllers/subsystem/job.dm (hunk 2 above)
Comment (Crexfu) [inline code/controllers/subsystem/job.dm]: done

**Comment by coiax** [comment]
This appears to be fucking with people's job priorities big time.

**Comment by francinum** [comment]
Absolutely fucking broken.

**Pull request merged**